src/index.ts

```diff
@@ -143,9 +143,42 @@ export class WalletPluginAnchor implements WalletPlugin {
      */
     // eslint-disable-next-line @typescript-eslint/no-unused-vars
     async sign(resolved: ResolvedSigningRequest, context: TransactContext): Promise<Signature> {
+        // Create the identity request to be presented to the user
+        const {callback, request} = await createTransactionRequest(context)
+
+        // Tell Wharf we need to prompt the user with a QR code and a button
+        context.ui.prompt({
+            title: 'Sign transaction in Anchor',
+            body: 'Scan the QR-code with Anchor on another device or use the button to open it here.',
+            elements: [
+                {
+                    type: 'qr',
+                    data: String(request),
+                },
+                {
+                    type: 'button',
+                    label: 'Open Anchor',
+                    data: String(request),
+                },
+            ],
+        })
+
+        if (session) {
+            session.transact(request)
+        }
```

Comment:
This whole plugin is getting called from within a `session.transact` call:

https://github.com/wharfkit/session/blob/1fa5202f37a1e61c61f08adc1d855889b6ee2148/src/session.ts#L468-L470

So it shouldn't be calling `transact` at all, otherwise it's going to cause an infinite loop 😂

Reply:
lol yep.. I added this code block because I figured that we would want to handle both in session and out of session `transact` calls. After thinking about it more, I realize that it really doesn't make sense to have that logic here 😅  I do think that it would be cool to have a static `Session.transact` method that dApps can call to just trigger Anchor transactions without the need for a session though.